File: docs/guides/php/src/useConditionlessRule.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\Consequence;
use Algolia\AlgoliaSearch\Model\Search\Rule;
use Algolia\AlgoliaSearch\Model\Search\TimeRange;

try {
    $client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

    $objectID = 'a-rule-id';

    $rule = (new Rule())
        ->setObjectID($objectID)
        ->setConsequence(new Consequence(/* Set relevant consequence */))
    ;

    // Set validity (optional)
    $rule->setValidity(
        [
            (new TimeRange())
                ->setFrom(1688774400)
                ->setUntil(1738972800),
        ]
    );

    $client->saveRule(
        '<YOUR_INDEX_NAME>',
        $objectID,
        $rule,
    );
} catch (Exception $e) {
    echo $e->getMessage().PHP_EOL;
}

File: docs/guides/ruby/useConditionlessRule.rb
require "algolia"

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

object_id = "a-rule-id"

rule = Algolia::Search::Rule.new(
  object_id: object_id,
  consequence: Algolia::Search::Consequence.new
  # Set relevant consequence
)

# Set validity (optional)
rule.validity = [
  Algolia::Search::TimeRange.new(
    from: 1688774400,
    until: 1738972800
  )
]

client.save_rule("<YOUR_INDEX_NAME>", algolia_object_id, rule)
